fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -230,21 +227,30 @@ def to(  # type: ignore
         .. note::
             This method modifies the module in-place.
 
+        .. warning:
+            Device changes are not supported, and this will raise an exception. The issue in that case is not
```

Comment:
nice warning. if possible, perhaps an assert to catch that?

Reply:
there is :) check the .to() method